staging/src/k8s.io/client-go/util/workqueue/queue.go

```diff
@@ -23,60 +23,64 @@ import (
 	"k8s.io/utils/clock"
 )
 
-type Interface interface {
-	Add(item interface{})
+type Interface TypedInterface[any]
+
+type TypedInterface[T comparable] interface {
+	Add(item T)
 	Len() int
-	Get() (item interface{}, shutdown bool)
-	Done(item interface{})
+	Get() (item T, shutdown bool)
+	Done(item T)
 	ShutDown()
 	ShutDownWithDrain()
 	ShuttingDown() bool
 }
 
 // Queue is the underlying storage for items. The functions below are always
 // called from the same goroutine.
-type Queue interface {
+type Queue[T comparable] interface {
 	// Touch can be hooked when an existing item is added again. This may be
 	// useful if the implementation allows priority change for the given item.
-	Touch(item interface{})
+	Touch(item T)
 	// Push adds a new item.
-	Push(item interface{})
+	Push(item T)
 	// Len tells the total number of items.
 	Len() int
 	// Pop retrieves an item.
-	Pop() (item interface{})
+	Pop() (item T)
 }
 
 // DefaultQueue is a slice based FIFO queue.
-func DefaultQueue() Queue {
-	return new(queue)
+func DefaultQueue[T comparable]() Queue[T] {
```

Comment:
Does this also need the `Typed` approach?

Reply:
Same as the above